src/transformers/generation/utils.py

```diff
@@ -1087,6 +1087,24 @@ def _validate_model_class(self):
                 exception_message += f" Please use one of the following classes instead: {generate_compatible_classes}"
             raise TypeError(exception_message)
 
+    def _validate_assistant(self, assistant_model):
+        if assistant_model is not None:
+            if type(self).__name__ in MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING._model_mapping.values():
+                if type(assistant_model).__name__ in MODEL_FOR_CAUSAL_LM_MAPPING._model_mapping.values():
```

Comment:
Would it be more general if we could simply check: `self.config.is_encoder_decoder` and `assistant_model.config.is_encoder_decoder`? If both True, then we know both are enc-dec models. If True and False, then the assistant is dec-only.

Reply:
Could it be that the assistant is encoder only ? in this case it would be perhaps better to check that `assistant_model.config.is_decoder`. 

However, when using whisper models with `AutoModelForCausalLM`, `self.config.is_decoder` is False. Is that expected ?

Reply:
I can make the changes with `self.config.is_encoder_decoder` and `assistant_model.config.is_encoder_decoder` for now.